File: Geek Assistant (Android AIO)/Forms/info.vb
Class info

    Public info_IconLight As Image = Nothing
    Public info_IconDark As Image = Nothing
    Public info_TextColorLight As Color = Nothing
    Public info_TextColorDark As Color = Nothing

    Private Sub info_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        GA_SetTheme.Run(Name)

        'Reset info answer
        infoAnswer = False
        No_Button.Select()

        If My.Settings.info_MsgTitle Is Nothing Or My.Settings.info_Msg Is Nothing Or My.Settings.info_MsgLevel < -1 Then
            DoMsg(10, $"Where's the error message??{vbCrLf}If you see this please contact the developer to fix this bug.{vbCrLf}Please provide the log when reporting.", 2)
            Close()
        End If

        'Reset to default
        No_Button.Text = "Close"
        Yes_Button.Text = "(Yes)"
        'GeekAssistant_PictureBox.Visible = False
        Yes_Button.Visible = False
        Copy_Button.Visible = True

        Dim msglevelText As String = "Warning"
        If My.Settings.info_MsgLevel = 1 Or My.Settings.info_MsgLevel = 10 Then
            msglevelText = "Error"
            If My.Settings.DarkTheme Then
                info_PictureBox.Image = My.Resources.Warning_Red_dark_64
                title_Label.ForeColor = Color.FromArgb(255, 191, 191)
            Else
                info_PictureBox.Image = My.Resources.Warning_Red_64
                title_Label.ForeColor = Color.FromArgb(154, 0, 0)
            End If
        ElseIf My.Settings.info_MsgLevel = 2 Then
            'msglevelText = "Question"
            No_Button.Text = infoButtonText.RightButton
            Yes_Button.Text = infoButtonText.LeftButton
            Yes_Button.Visible = True
            Text = $"{My.Settings.info_MsgTitle} — Geek Assistant"
            title_Label.Text = My.Settings.info_MsgTitle
            'GeekAssistant_PictureBox.Visible = True
            Copy_Button.Visible = False
            If My.Settings.DarkTheme Then
                info_PictureBox.Image = My.Resources.Question_Blue_Dark_64
                title_Label.ForeColor = Color.FromArgb(186, 221, 253)
            Else
                info_PictureBox.Image = My.Resources.Question_Blue_64
                title_Label.ForeColor = Color.FromArgb(64, 109, 128)
            End If
        Else
            If My.Settings.DarkTheme Then
                info_PictureBox.Image = My.Resources.Info_Yellow_dark_64
                title_Label.ForeColor = Color.FromArgb(255, 238, 191)
            Else
                info_PictureBox.Image = My.Resources.Info_Yellow_64
                title_Label.ForeColor = Color.FromArgb(115, 84, 0)
            End If
        End If

        ''''Special Cases
        If title_Label.Text = "Send Feedback" Then
            If My.Settings.DarkTheme Then
                info_PictureBox.Image = My.Resources.Smile_dark_64
                title_Label.ForeColor = Color.FromArgb(191, 255, 191)
            Else
                info_PictureBox.Image = My.Resources.Smile_64
                title_Label.ForeColor = Color.FromArgb(0, 102, 71)
            End If
        End If

        If My.Settings.info_MsgLevel <> 2 Then   'Avoid when it is a question
            Text = $"{msgIcon}{msglevelText}: ❰{My.Settings.ErrorCode}❱ — Geek Assistant"
            title_Label.Text = $"❰{My.Settings.ErrorCode}❱ {My.Settings.info_MsgTitle}"
        End If
        msg_Textbox.Text = My.Settings.info_Msg

        Width = 566
        Height = 355
        Select Case msg_Textbox.Text.Length
            Case < 150
                Width = 520
                Height -= 50
            Case > 400
                Width = 620
                Height += 50
        End Select
        SetBounds((Main.Width / 2) - (Width / 2) + Main.Location.X, Main.Top, Width, Height)
    End Sub

    Private Sub OK_Button_Mousedown(sender As Object, e As EventArgs) Handles Yes_Button.MouseDown
        Yes_Button.ForeColor = Color.White
    End Sub
    Private Sub OK_Button_MouseUp(sender As Object, e As EventArgs) Handles Yes_Button.MouseUp
        Yes_Button.ForeColor = SystemColors.Highlight
    End Sub
    Private Sub OK_Button_Click(sender As Object, e As EventArgs) Handles Yes_Button.Click
        If My.Settings.info_MsgLevel = 2 Then
            infoAnswer = True
            Close()
        Else
            Process.Start("https://developer.android.com/studio/run/oem-usb")
        End If
    End Sub

    'Private Sub Close_Button_MouseDown(sender As Object, e As EventArgs) Handles Close_Button.MouseDown
    '    Close_Button.ForeColor = Color.White
    'End Sub
    'Private Sub Close_Button_MouseUp(sender As Object, e As EventArgs) Handles Close_Button.MouseUp
    '    Close_Button.ForeColor = SystemColors.ControlText
    'End Sub
    Private Sub Close_Button_Click(sender As Object, e As EventArgs) Handles No_Button.Click
        If Not My.Settings.VerboseLogging And My.Settings.VerboseLoggingPrompt _
            And My.Settings.DeviceAPILevel <> 2 And Not title_Label.Text = "Enable Verbose Logging?" And Not title_Label.Text = "Send Feedback" Then
            'Close()
            Dispose()
            Main.BringToFront()
            If GA_infoAsk.Run("Enable Verbose Logging?",
                              "For easier troubleshooting, enable verbose logging in Settings and try again",
                              "Enable Verbose Logging", "Close") Then
                GA_Settings.ShowDialog()
            End If
            My.Settings.VerboseLoggingPrompt = False
        End If
        If My.Settings.info_MsgLevel = 2 Then
            infoAnswer = False
        End If
        Close()
        'Dispose()
    End Sub

    Private Sub CopyToClipboard_MouseDown(sender As Object, e As EventArgs) Handles Copy_Button.MouseDown
        Copy_Button.Image = My.Resources.Copy_W_24
        Copy_Button.ForeColor = Color.White
    End Sub
    Private Sub CopyToClipboard_MouseUp(sender As Object, e As EventArgs) Handles Copy_Button.MouseUp
        Copy_Button.Image = My.Resources.Copy_B_24
        If CopyToClipboard_Timer.Enabled Then
            Copy_Button.ForeColor = Color.DarkGreen
        Else Copy_Button.ForeColor = SystemColors.ControlText
        End If
    End Sub
    Private Sub CopyToClipboard_Click(sender As Object, e As EventArgs) Handles Copy_Button.Click
        CopyToClipboard_Timer.Enabled = False
        Clipboard.SetText(title_Label.Text & vbCrLf & vbCrLf & msg_Textbox.Text)
        Copy_Button.ForeColor = Color.DarkGreen
        Copy_Button.Text = "Copied "
        CopyToClipboard_Timer.Enabled = True
    End Sub

    Private Sub CopyToClipboard_Timer_Tick(sender As Object, e As EventArgs) Handles CopyToClipboard_Timer.Tick
        CopyToClipboard_Timer.Enabled = False
        Copy_Button.Text = "Copy  "
        Copy_Button.ForeColor = SystemColors.ControlText
    End Sub

    Private Sub title_Label_TextChanged(sender As Object, e As EventArgs) Handles title_Label.TextChanged
        If title_Label.Text.Length < 25 Then
            title_Label.Font = New Font("Segoe UI", 15.75!, FontStyle.Regular, GraphicsUnit.Point, 0)
        Else
            title_Label.Font = New Font("Segoe UI", 12.0!, FontStyle.Regular, GraphicsUnit.Point, 0)
        End If
    End Sub

    Private WithEvents title_Label_Click_Timer As New Timer With {.Interval = 1500}
    Private savedTitle As String
    Private CopiedText As String = "Copied Error info..."
    Private Sub title_Label_Click(sender As Object, e As EventArgs) Handles title_Label.Click
        If title_Label.Text = CopiedText Then Exit Sub
        If My.Settings.info_MsgLevel <> 2 Then
            savedTitle = title_Label.Text
            title_Label.Text = CopiedText
            Clipboard.SetText(savedTitle)
            title_Label_Click_Timer.Enabled = True
        End If
    End Sub
    Private Sub title_Label_Click_Timer_Tick() Handles title_Label_Click_Timer.Tick
        title_Label.Text = savedTitle
        title_Label_Click_Timer.Enabled = False
    End Sub
End Class

File: Geek Assistant (Android AIO)/Forms/Info.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GeekAssistant.Forms {
    public partial class Info : Form {
        public Info() {
            InitializeComponent();
        }
        private void AssignEvents() {
            Yes_Button.MouseDown += Yes_Button_Mousedown;
            Yes_Button.MouseUp += Yes_Button_MouseUp;
            Yes_Button.Click += Yes_Button_Click;

            No_Button.Click += No_Button_Click;

            Copy_Button.MouseDown += Copy_Button_MouseDown;
            Copy_Button.MouseUp += Copy_Button_MouseUp;
            Copy_Button.Click += Copy_Button_Click;

            CopyToClipboard_Timer.Tick += CopyToClipboard_Timer_Tick;

            title_Label.TextChanged += title_Label_TextChanged;
            title_Label.Click += title_Label_Click;
            title_Label_Click_Timer.Tick += title_Label_Click_Timer_Tick;
        }

        public Image info_IconLight;
        public Image info_IconDark;
        public Color info_TextColorLight;
        public Color info_TextColorDark;

        private void Info_Load(object sender, EventArgs e) {
            AssignEvents();

            //Reset info answer
            GA_infoAsk.infoAnswer = false;
            No_Button.Select();

            if (common.S.info_MsgTitle == null || common.S.info_Msg == null || common.S.info_MsgLevel < -1) {
                Dispose(); //dispose before starting again
                GA_Msg.DoMsg(10, $"Where//s the error message??\n" +
                          "if ( you see this please contact the developer to fix this bug.\nPlease provide the log when reporting.", 2);
            }

            //Reset to default
            No_Button.Text = "Close";
            Yes_Button.Text = "Yes";
            //GeekAssistant_PictureBox.Visible = false;
            Yes_Button.Visible = false;
            Copy_Button.Visible = true;

            string msglevelText = "Warning";
            if (common.S.info_MsgLevel == 1 | common.S.info_MsgLevel == 10) {
                msglevelText = "Error";
                if (common.S.DarkTheme) {
                    info_PictureBox.Image = prop.x64.Warning_Red_dark_64;
                    title_Label.ForeColor = Color.FromArgb(255, 191, 191);
                } else {
                    info_PictureBox.Image = prop.x64.Warning_Red_64;
                    title_Label.ForeColor = Color.FromArgb(154, 0, 0);
                }
            } else if (common.S.info_MsgLevel == 2) {
                //msglevelText = "Question"
                No_Button.Text = GA_infoAsk.infoButtonText.RightButton;
                Yes_Button.Text = GA_infoAsk.infoButtonText.LeftButton;
                Yes_Button.Visible = true;
                Text = $"{common.S.info_MsgTitle} — Geek Assistant";
                title_Label.Text = common.S.info_MsgTitle;
                //GeekAssistant_PictureBox.Visible = true;
                Copy_Button.Visible = false;
                if (common.S.DarkTheme) {
                    info_PictureBox.Image = prop.x64.Question_Blue_Dark_64;
                    title_Label.ForeColor = Color.FromArgb(186, 221, 253);
                } else {
                    info_PictureBox.Image = prop.x64.Question_Blue_64;
                    title_Label.ForeColor = Color.FromArgb(64, 109, 128);
                }
            } else {
                if (common.S.DarkTheme) {
                    info_PictureBox.Image = prop.x64.Info_Yellow_dark_64;
                    title_Label.ForeColor = Color.FromArgb(255, 238, 191);
                } else {
                    info_PictureBox.Image = prop.x64.Info_Yellow_64;
                    title_Label.ForeColor = Color.FromArgb(115, 84, 0);
                }
            }


            if (common.S.info_MsgLevel != 2) {   //Avoid when it is a question
                Text = $"{GA_Msg.msgIcon}{msglevelText}: ❰{common.ErrorInfo.code}❱ — Geek Assistant";
                title_Label.Text = $"❰{common.ErrorInfo.code}❱ {common.S.info_MsgTitle}";
            }
            msg_Textbox.Text = common.S.info_Msg;

            ////////Special Cases
            if (title_Label.Text == "Send Feedback") {
                if (common.S.DarkTheme) {
                    info_PictureBox.Image = prop.x64.Smile_dark_64;
                    title_Label.ForeColor = Color.FromArgb(191, 255, 191);
                } else {
                    info_PictureBox.Image = prop.x64.Smile_64;
                    title_Label.ForeColor = Color.FromArgb(0, 102, 71);
                }
            }

            //// Set size
            Width = 566;
            Height = 355;
            switch (msg_Textbox.Text.Length) {
            case < 150:
                Width = 520;
                Height -= 50;
                break;
            case > 400:
                Width += 60;
                Height += 75;
                break;
            }
            SetBounds((common.Home.Width / 2) - (Width / 2) + common.Home.Location.X, common.Home.Top, Width, Height);


            GA_SetTheme.Run(Name);
        }

        private void Yes_Button_Mousedown(object sender, EventArgs e) {
            if (Yes_Button.ForeColor.GetBrightness() >= .5)
                Yes_Button.ForeColor = Color.Black;
            else Yes_Button.ForeColor = Color.White;

        }
        private void Yes_Button_MouseUp(object sender, EventArgs e) {
            Yes_Button.ForeColor = title_Label.ForeColor;
        }
        private void Yes_Button_Click(object sender, EventArgs e) {
            if (common.S.info_MsgLevel == 2) {
                GA_infoAsk.infoAnswer = true;
                Close();
            }
        }

        //private void Close_Button_MouseDown(object sender, EventArgs e) { Close_Button.MouseDown
        //    Close_Button.ForeColor = Color.White
        //}
        //private void Close_Button_MouseUp(object sender, EventArgs e) { Close_Button.MouseUp
        //    Close_Button.ForeColor = SystemColors.ControlText
        //}
        private void No_Button_Click(object sender, EventArgs e) {
            if (!common.S.VerboseLogging && common.S.VerboseLoggingPrompt
            && common.S.info_MsgLevel != 2) { //&& ! title_Label.Text = "Enable Verbose Logging?" && ! title_Label.Text = "Send Feedback" ) {
                                              //Close();
                Dispose();
                if (GA_infoAsk.Run("Enable Verbose Logging?",
                                   $"For better troubleshooting, enable verbose logging in Settings and try again.",
                                  "Enable", "Close"))
                    common.Settings.ShowDialog();

                common.S.VerboseLoggingPrompt = false;
            }
            if (common.S.info_MsgLevel == 2) GA_infoAsk.infoAnswer = false;
            Close();
            common.Home.BringToFront();
            //Dispose();
        }

        private void Copy_Button_MouseDown(object sender, EventArgs e) {
            Copy_Button.Image = prop.x24.Copy_W_24;
            Copy_Button.ForeColor = Color.White;
        }
        private void Copy_Button_MouseUp(object sender, EventArgs e) {
            Copy_Button.Image = prop.x24.Copy_B_24;
            if (CopyToClipboard_Timer.Enabled)
                Copy_Button.ForeColor = Color.DarkGreen;
            else Copy_Button.ForeColor = SystemColors.ControlText;

        }
        private void Copy_Button_Click(object sender, EventArgs e) {
            CopyToClipboard_Timer.Enabled = false;
            Clipboard.SetText($"{title_Label.Text}\n\n{msg_Textbox.Text}");
            Copy_Button.ForeColor = Color.DarkGreen;
            Copy_Button.Text = "Copied ";
            CopyToClipboard_Timer.Enabled = true;
        }

        private void CopyToClipboard_Timer_Tick(object sender, EventArgs e) {
            CopyToClipboard_Timer.Enabled = false;
            Copy_Button.Text = "Copy  ";
            Copy_Button.ForeColor = SystemColors.ControlText;
        }

        private void title_Label_TextChanged(object sender, EventArgs e) {
            if (title_Label.Text.Length < 25)
                title_Label.Font = new Font("Segoe UI", 15.75F, FontStyle.Regular, GraphicsUnit.Point, 0);
            else title_Label.Font = new Font("Segoe UI", 12.0F, FontStyle.Regular, GraphicsUnit.Point, 0);

        }

        private Timer title_Label_Click_Timer = new Timer { Interval = 1500 };
        private string savedTitle;
        private string CopiedText = "Copied information...";
        private void title_Label_Click(object sender, EventArgs e) {
            if (title_Label.Text == CopiedText) return;
            if (common.S.info_MsgLevel != 2) {
                savedTitle = title_Label.Text;
                title_Label.Text = CopiedText;
                Clipboard.SetText(savedTitle);
                title_Label_Click_Timer.Start();
            }
        }
        private void title_Label_Click_Timer_Tick(object sender, EventArgs e) {
            title_Label.Text = savedTitle;
            title_Label_Click_Timer.Stop();
        }

    }
}
